ui/src/main/java/com/xmartlabs/bigbang/ui/common/recyclerview/BaseRecyclerViewAdapter.java

```diff
@@ -227,11 +229,47 @@ public void removeItems(@NonNull List<Object> items) {
         });
   }
 
+  /**
+   * Sets the items data for the recycler view and notifying any registered observers that the data set has
+   * changed. It uses a function that calculates the difference between the old and the new items
+   * in order to improve the update process.
+   *
+   * @param <T>                       Type of the items to be added.
+   * @param newItems                  Items to be added. Each Pair consists of an item and its RecycleItemType.
+   * @param areItemsTheSameFunction   A function which checks that two items are the same.
+   * @param areContentTheSameFunction A function which checks that the content of two items are the same.
+   */
+  protected <T> void setMultipleTypeItems(final @Nullable List<Pair<? extends RecycleItemType,T>> newItems,
+                                          @NonNull BiFunction<T, T, Boolean> areItemsTheSameFunction,
+                                          @NonNull BiFunction<T, T, Boolean> areContentTheSameFunction) {
+    if (CollectionHelper.isNullOrEmpty(newItems)) {
```

Comment:
this is an issue, you should call clear all. Can you fix it @FerrariBruno ?

Reply:
yup

Reply:
cool! ty